pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -726,12 +730,19 @@ func getOSDInfo(d *apps.Deployment) ([]OSDInfo, error) {
 		}
 		locationPrefix := "--crush-location="
 		if strings.HasPrefix(a, locationPrefix) {
+			locationFound = true
 			// Extract the same CRUSH location as originally determined by the OSD prepare pod
 			// by cutting off the prefix: --crush-location=
 			osd.Location = a[len(locationPrefix):]
 		}
 	}
 
+	if !locationFound {
+		pods, _ := c.context.Clientset.CoreV1().Pods(d.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%v=%d", OsdIdLabelKey, osd.ID)})
+		hostName, _ := k8sutil.GetNodeHostName(c.context.Clientset, pods.Items[0].Spec.NodeName)
```

Comment:
What is the err returned from this line and the previous line? If it's not working yet these will probably help troubleshoot.